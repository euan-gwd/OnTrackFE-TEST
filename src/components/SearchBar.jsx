import React, { useState } from 'react';
import { connect } from 'react-redux';
import { getBooks } from '../store/actions';
import InputGroup from 'react-bootstrap/InputGroup';
import Form from 'react-bootstrap/Form';
import FormControl from 'react-bootstrap/FormControl';
import Button from 'react-bootstrap/Button';

export const SearchBar = ({ dispatch, page, itemsPerPage }) => {
  const [searchTerm, setSearchTerm] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();

    const pageOptions = {
      page: page,
      itemsPerPage: itemsPerPage,
      filters: [{ type: 'all', values: [searchTerm] }]
    };

    dispatch(getBooks(pageOptions));
  };

  return (
    <Form noValidate onSubmit={handleSubmit}>
      <InputGroup size="sm" className="mb-3">
        <FormControl
          aria-label="Small"
          aria-describedby="inputGroup-sizing-sm"
          value={searchTerm}
          onChange={(e) => setSearchTerm(e.target.value)}
        />
        <InputGroup.Append>
          <Button id="inputGroup-sizing-sm" variant="primary" type="submit">
            Search
          </Button>
        </InputGroup.Append>
      </InputGroup>
    </Form>
  );
};

const mapStateToProps = (state) => ({
  page: state.page,
  itemsPerPage: state.itemsPerPage
});

export default connect(mapStateToProps)(SearchBar);
