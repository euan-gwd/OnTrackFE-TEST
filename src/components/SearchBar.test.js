import React from 'react';
import { shallow } from 'enzyme';
import { SearchBar } from './SearchBar';

describe('SearchBar', () => {
  it('renders as expected', () => {
    const props = {
      page: 1,
      itemsPerPage: 20
    };

    const wrapper = shallow(<SearchBar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
