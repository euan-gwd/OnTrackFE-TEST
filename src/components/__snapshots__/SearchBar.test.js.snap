// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders as expected 1`] = `
<Form
  inline={false}
  noValidate={true}
  onSubmit={[Function]}
>
  <InputGroup
    className="mb-3"
    size="sm"
  >
    <FormControl
      aria-describedby="inputGroup-sizing-sm"
      aria-label="Small"
      onChange={[Function]}
      value=""
    />
    <InputGroupAppend>
      <Button
        active={false}
        disabled={false}
        id="inputGroup-sizing-sm"
        type="submit"
        variant="primary"
      >
        Search
      </Button>
    </InputGroupAppend>
  </InputGroup>
</Form>
`;
